core/src/main/scala/org/scalajs/macrotaskexecutor/MacrotaskExecutor.scala

```diff
@@ -67,17 +66,15 @@ object MacrotaskExecutor extends ExecutionContextExecutor {
         if (currentlyRunningATask) {
           js.Dynamic.global.setTimeout(() => runIfPresent(handle), 0)
         } else {
-          tasksByHandle.get(handle) match {
-            case Some(task) =>
-              currentlyRunningATask = true
-              try {
-                task()
-              } finally {
-                tasksByHandle -= handle
-                currentlyRunningATask = false
-              }
-
-            case None =>
+          val task = tasksByHandle(handle).asInstanceOf[js.UndefOr[Runnable]]
```

Comment:
I'm pretty sure you need `tasksByHandle.selectDynamic(handle)` here. As is, you're *calling* `tasksByHandle` with `handle` as an argument.

Reply:
Yes, you're right ... messed up the refactor from an initial try with `@BracketAccess`. But `selectDynamic` requires a string, so going to try `@BracketAccess` again